src/common_upgrades/change_macro_in_globals.py

```diff
@@ -0,0 +1,249 @@
+
+        """
+        if not type(macro_change) == list:
+            macro_change = [macro_change, ]
+
+        for macro in macro_change:
+            macro = self._sanitise_input_dictionary_tuples(macro)
+
+            for index in self._globals_filter_generator(macro["ioc_name"]):
+                self._apply_regex_macro_change(macro, index)
+
+        self.write_modified_globals_file()
+
+        return None
+
+    def _globals_filter_generator(self, ioc_to_change):
+        """
+        Returns lines containing specified IOCs from globals.txt
+
+        Generator that gives all the lines for a given IOC in globals.txt and saves them back to their original location
+        after they've been yielded. This will match IOCs with the same name as the root plus any that have a number
+        appended in the form _XX. To change the line change the yielded lines[index] to the value given.
+
+        Args:
+            ioc_to_change: the root name of the ioc to change
+        """
+        for index, line in enumerate(self._loaded_file if self._loaded_file is not None else []):
+            if line.startswith("{}_".format(ioc_to_change)):
+                self._logger.info("Found line '{}' in {}".format(line, GLOBALS_FILENAME))
+                yield index
+
+    def _dict_field_to_tuple(self, dict_field):
+        """
+        Converts a string to a 1-tuple containing that string, or returns a 1- or 2-tuple input
+        Args:
+            dict_field: A string to be contained in a 1-tuple, or a tuple
+
+        Returns:
+            field_as_tuple: tuple representation of dict_field, if it is not already a tuple
+        """
+
+        if type(dict_field) is not tuple:
+            field_as_tuple = (dict_field,)
+        else:
+            field_as_tuple = dict_field
+
+        return field_as_tuple
+
+    def _sanitise_input_dictionary_tuples(self, macro_change):
+        """
+        Standardises the macros which have been user inputted into tuples of length 1 or 2.
+
+        Args:
+            macro_change: A dict-like object containing three fields:
+            ioc_name : the IOC name,
+            old_macro: old_macro_name or 2-tuple of (old_macro_name, old_macro_value)
+            new_macro: new_macro_name or 2-tuple of (new_macro_name, new_macro_value)
+
+        Returns:
+            sanitised_macro_change: A dict-like object containing three fields:
+            ioc_name : the IOC name,
+            old_macro: tuple of (old_macro_name,) or (old_macro_name, old_macro_value)
+            new_macro: tuple of (new_macro_name) or (new_macro_name, new_macro_value)
+
+
+        """
+        sanitised_macro_change = macro_change.copy()
+
+        sanitised_macro_change["new_macro"] = self._dict_field_to_tuple(macro_change["new_macro"])
+        sanitised_macro_change["old_macro"] = self._dict_field_to_tuple(macro_change["old_macro"])
+
+        return sanitised_macro_change
+
+    def _determine_filled_fields(self, macro):
```

Comment:
I'm a bit confused by this method, I think you can move setting the None to the _dict_field_to_tuple method. Otherwise all this does is split the tuple and put it back together again?